### ISSUE TITLE
WAGO 750-881

### ISSUE THREAD

**Issue by MichielVanwelsenaere** [SebastianRau/Codesys2-Mqtt-lib#2]
Hi,

I've tested this library on the WAGO 750-881 and haven't experienced any issues. If you wan't you can add it to the 'Tested on' section. Note that the util.lib library gave some compilation issues on the BLINK POU, i ended up resolving it by excluding the POU from the build.

Do you think it would be possible to reset the "spayload" string automatically to an empty string after it has been transmitted?

Very nice work, much appreciated!

**Comment by sebastianrau** [comment]
Hi,
thanks for testing. I'll check your point with sPayload and will add some Documentations and code Examples in the next days. Why exactly do you need the "empty" string. Maybe is the bPublish Variable what you need.. Here`s my implementation using a timer and the bPublish variable to send every "tSendTime" seconds.


```
fbTimerPublish	: TON;
fbFtrig_publish		: F_TRIG;
bMQTT_Publish 	: BOOL;
------------------------------------------------------
fbTimerPublish(IN:= bTimerStart , PT:=tSendTime , Q=> , ET=> );
fbFtrig_publish(CLK:=bMQTT_Publish);		(*Send Done detection*)

IF fbFtrig_publish.Q THEN
	bTimerStart := TRUE;	(*Start Timer when sending is done *)
END_IF

mqttClient(
	sTopic:= sMQTT_Topic,
	sPayload:= sMQTT_Payload,
	sIp:= '**********' ,
	sClientID := '************',
	iPortNumber:= 1883,
	xPublish:= bMQTT_Publish,
	xConnected=> ,
	xBusy=> );

IF fbTimerPublish.Q AND NOT bMQTT_Publish THEN    (*Timer ended and not publishing*)
	bTimerStart :=  FALSE;		(* Reset Timer *)

	(* CREATE YOUR PAYLOAD STRING	*)
	sMQTT_Payload := sPayload;
	bMQTT_Publish := TRUE;
END_IF
```
Thanks for your input!

Sebastian

**Comment by MichielVanwelsenaere** [comment]
Hi,
thanks for the response. In my logic i'm filling up the payload string using concat. Depending on certain event flags data is added untill it is full or the end of the cycle has been reached. In a new cycle data can again be transmitted depending on those certain event flags. as the old is to be considered obsolete I manually reset the string in the beginning of each cycle. 
 
It would be a bit more "maintainable" if the string was reset automatically each new cycle. Yet this would introduce a breaking change so I would understand if request would be rejected :-)

Thanks for the code example above!

**Comment by sebastianrau** [comment]
Hi,
TCP/IP Communication is an asynchronous communication. It's possible, that your PLC cycle is much faster than your communication via MQTT. 
For example, if you have a slow Internet connection with a bad Ping-Time (e.g. 100ms) and your PLC has a cyclic Task-Time of 10ms. In this case trying to send 20-times faster than your Internet connection can transfer a TCP/IP communication (it's always Request and Response -> 100ms (Req) + 100ms (Resp.) = 200ms).
If you now try to send every cycle you'll produce a big traffic yam, because your TCP/IP connection is not able to send the data in that speed.
A simple buffer will not help, because on a cyclic task you don`t a "free" time to send the packtes out of the buffer, it will overrun really fast.

In short, your plan is not useful for a asynchronous communication. The only way to handle it, is to wait until the transfer is complete and send the next after that.

Regards
Sebastian
